File: blockchain/blockchain.py
from block import *

class Blockchain:
    def __init__(self, chain):
        self.chain = chain
    
    def addBlock(self, data):
        newBlock = mineBlock(self.chain[len(self.chain)-1], data)
        self.chain.append(newBlock)
        return newBlock

def isValidChain(chain):
    if (chain[0].hash != "genesisHash"):
        return False
    
    for i in range(1, len(chain)):
        block = chain[i]
        lastBlock = chain[i-1]
        if (block.lashHash != lastBlock.hash or block.hash != blockHash(block)):
            return False
        
    return True

File: old/blockchain.js
const Block = require("./block.js");

class Blockchain {
  constructor(chain) {
    this.chain = chain;
  }

  /**
   * Given data, mine a new block with the particular data and push it to the chain
   * @param {string} data
   * @returns {Block}
   */
  addBlock(data) {
    const minedBlock = Block.mineBlock(this.chain[this.chain.length - 1], data);
    this.chain.push(minedBlock);
    return minedBlock;
  }

  /**
   * Given a chain, return true if it is valid, else return false
   * @param {array} chain
   * @returns {boolean}
   */
  isValidChain(chain) {
    //base case
    if (chain[0].hash !== "genesisHash") return false;

    for (let i = 1; i < chain.length; i++) {
      const block = chain[i];
      const lastBlock = chain[i - 1];
      if (block.lastHash !== lastBlock.hash || block.hash !== Block.blockHash(block)){
        return false;
      }
    }
    return true;
  }
}

module.exports = Blockchain;
